components/script/dom/node.rs

```diff
@@ -771,6 +771,20 @@ impl Node {
         self.owner_doc().is_html_document()
     }
 
+    // https://dom.spec.whatwg.org/#in-a-document-tree
+    pub fn is_in_a_document_tree(&self) -> bool {
```

Comment:
The problem is that `is_in_doc` returns true if the node is in a document fragment.